Sources/Property.swift

```diff
@@ -700,18 +701,21 @@ public final class MutableProperty<Value>: ComposableMutablePropertyProtocol {
 /// The requirement of a `Value?` storage from composed properties prevents further
 /// implementation sharing with `MutableProperty`.
 private final class PropertyBox<Value> {
-	let lock = NSRecursiveLock()
-	var value: Value
+	private let lock = NSRecursiveLock()
+	private var _value: Value
+	var value: Value { return modify { $0 } }
 
 	init(_ value: Value) {
-		self.value = value
+		self._value = value
+	}
+
+	func modify<Result>(_ action: (inout Value) throws -> Result) rethrows -> Result {
+		return try modifyDidSet(action, didSet: nil)
 	}
-}
 
-extension NSRecursiveLock {
-	fileprivate func sync<Result>(_ action: () throws -> Result) rethrows -> Result {
-		lock(); defer { unlock() }
-		return try action()
+	func modifyDidSet<Result>(_ action: (inout Value) throws -> Result, didSet: (() -> Void)?) rethrows -> Result {
```

Comment:
Ah, I thought the trailing closure would be in conflict. But I just realize `didSet` can be moved as the first. 😛